Bottom: 2de895208ca32b10f2a0f8c03518a99490dc55a7
Top:    4837a7252f09bf23449c7cba09f0cb540d15f32e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-13 19:18:29 +0100

PCB+GL: Fix GL headers for Win32


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 7a348c8..0347b29 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -32,12 +32,18 @@
 #include <math.h>
 #include <assert.h>
 
-/* The Linux OpenGL ABI 1.0 spec requires that we define
- * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
- * in order to get prototypes:
- *   http://www.opengl.org/registry/ABI/
- */
-#define GL_GLEXT_PROTOTYPES 1
+#ifdef WIN32
+#   define WIN32_LEAN_AND_MEAN 1
+#   include <windows.h>
+#else
+    /* The Linux OpenGL ABI 1.0 spec requires that we define
+     * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
+     * in order to get prototypes:
+     *   http://www.opengl.org/registry/ABI/
+     */
+#   define GL_GLEXT_PROTOTYPES 1
+#endif
+
 #ifdef HAVE_OPENGL_GL_H
 #   include <OpenGL/gl.h>
 #else
@@ -67,7 +73,6 @@
 #include <dmalloc.h>
 #endif
 
-
 triangle_buffer buffer;
 float global_depth = 0;
